### PR TITLE
Feature: Added support for socks proxy via HTTP_PROXY env variable

### PR DESCRIPTION
**Pull request by pnoltof**
Added support for socks proxy via HTTP_PROXY env variable

### Motivation & Context
Using a SOCKS proxy for cifuzz remote-run was currently not possible.
These changes enable users to configure a SOCKS Proxy via the HTTP_PROXY environment variable.

### Description
The added dialer picks up any configured proxy settings. If no proxy settings are configured it will to a direct connection.

### CODE REVIEW

**Comment by jochil** [comment]
Thanks for the contribution. It seems like the checks for code formatting are failing... can you use our git hooks (https://github.com/CodeIntelligenceTesting/cifuzz/tree/main/githooks) or run `make fmt`?

File: internal/api/api.go [hunk 1]
```diff
@@ -301,3 +303,11 @@ func ValidateURL(s string) error {
 	}
 	return nil
 }
+
+func GetCustTransportLayer() *http.Transport {
+	dialer := proxy.FromEnvironment()
+	dialContext := func(ctx context.Context, network, address string) (net.Conn, error) {
+		return dialer.Dial(network, address)
+	}
+	return &http.Transport{DialContext: dialContext, DisableKeepAlives: true}
```
Comment (jochil) [inline internal/api/api.go:312]: What's the reason for `DisableKeepAlives`?

File: internal/api/api.go [hunk 2]
```diff
@@ -229,7 +231,7 @@ func (client *APIClient) sendRequest(method, endpoint, token string) (*http.Resp
 	}
 	req.Header.Add("Authorization", "Bearer "+token)
 
-	httpClient := &http.Client{}
+	httpClient := &http.Client{Transport: GetCustTransportLayer()}
```
Comment (jochil) [inline internal/api/api.go:234]: I think you don't have to use a custom function for this... this should work too
```go
proxyTransport := &http.Transport{ Proxy: http.ProxyFromEnvironment }
httpClient : = &http.Client{ Transport: proxyTransport }
```

**Comment by jochil** [comment]
> adjusted order of imports as requested by our code formatting checks
> changes done by go mod tidy

Some little addition... IMO you can squash these commits into the main one. It would help to keep the git log clean

File: internal/api/api.go (hunk 2 above)
Comment (pnoltof) [reply internal/api/api.go:234]: Sadly this doesn't work. The main reason for it being this: https://github.com/golang/go/issues/24135
It seams like the DNS lookups are not handled correctly with the default go http proxy.
This is why we needed the net.proxy and the Dialer.

File: internal/api/api.go [hunk 3]
```diff
@@ -256,3 +258,11 @@ func ValidateURL(s string) error {
 	}
 	return nil
 }
+
+func GetCustTransportLayer() *http.Transport {
+	dialer := proxy.FromEnvironment()
+	dialContext := func(ctx context.Context, network, address string) (net.Conn, error) {
+		return dialer.Dial(network, address)
+	}
+	return &http.Transport{DialContext: dialContext, DisableKeepAlives: true}
```
Comment (MarkusZoppelt) [inline internal/api/api.go:267]: Could you please add an explanatory comment to this function? I can't really tell what it does just from looking at it.

Also: Not sure if `Cust` is an acronym or just short for `Custom`.
If it is the latter, I vote for just calling it `GetCustomTransportLayer` or specifying what is "custom" in the function name if possible.

File: internal/api/api.go (hunk 1 above)
Comment (jochil) [reply internal/api/api.go]: removed

File: internal/api/api.go (hunk 3 above)
Comment (jochil) [reply internal/api/api.go]: I added a linked + explanation

**Review by MarkusZoppelt — APPROVED** [review]

**Review by jochil — APPROVED** [review]

**Pull request merged**